LibraryProject1/ProForma/ProFormaDescriptor.cs

```diff
@@ -1,29 +1,69 @@
-﻿namespace TestLibNamespace.ProForma
+﻿using System;
+
+namespace TestLibNamespace.ProForma
 {
     /// <summary>
     /// Member of the tag. Could be a key-value pair, or a keyless entry.
     /// </summary>
     public class ProFormaDescriptor
     {
+
+        #region Constructors
+
         /// <summary>
         /// Initializes a new instance of the <see cref="ProFormaDescriptor"/> class.
         /// </summary>
         /// <param name="key">The key.</param>
         /// <param name="value">The value.</param>
         public ProFormaDescriptor(string key, string value)
         {
-            this.Key = key;
+            if (Enum.TryParse(key, true, out ProFormaKey parsedKey))
+                this.Key = parsedKey;
+            else
+                throw new ProFormaParseException("The key " + key + " is not supported.");
+            this.Value = value;
+        }
+
+        /// <summary>
+        /// Initializes a descriptor without value only
+        /// </summary>
+        /// <param name="value"></param>
+        public ProFormaDescriptor(string value)
+        {
+            this.Key = ProFormaKey.MOD;
```

Comment:
I'd suggest that we set this to some 'Unknown' or 'Unspecified' enum value. This will allow the future ProForma writer to ignore it.

Reply:
The only key we don't require is mod, so I suggest we make a bool parameter for the writer that asks whether to write the optional mod key.